service/src/main/java/org/cbioportal/service/impl/StructuralVariantEnrichmentServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package org.cbioportal.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.cbioportal.model.AlterationEnrichment;
+import org.cbioportal.model.MolecularProfileCaseIdentifier;
+import org.cbioportal.model.StructuralVariantCountByGene;
+import org.cbioportal.service.StructuralVariantEnrichmentService;
+import org.cbioportal.service.StructuralVariantService;
+import org.cbioportal.service.exception.MolecularProfileNotFoundException;
+import org.cbioportal.service.util.AlterationEnrichmentUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class StructuralVariantEnrichmentServiceImpl implements StructuralVariantEnrichmentService {
+    @Autowired
+    private StructuralVariantService structuralVariantService;
+    @Autowired
+    private AlterationEnrichmentUtil<StructuralVariantCountByGene> alterationEnrichmentUtil;
+
+    @Override
+    public List<AlterationEnrichment> getStructuralVariantEnrichments(
+            Map<String, List<MolecularProfileCaseIdentifier>> molecularProfileCaseSets, String enrichmentType)
+            throws MolecularProfileNotFoundException {
+
+        Map<String, List<StructuralVariantCountByGene>> mutationCountsbyEntrezGeneIdAndGroup = molecularProfileCaseSets
+                .entrySet().stream().collect(Collectors.toMap(entry -> entry.getKey(), entry -> {
+                    // set value of each group to list of MutationCountByGene
+                    List<String> molecularProfileIds = new ArrayList<>();
+                    List<String> caseIds = new ArrayList<>();
+
+                    entry.getValue().forEach(molecularProfileCase -> {
+                        molecularProfileIds.add(molecularProfileCase.getMolecularProfileId());
+                        caseIds.add(molecularProfileCase.getCaseId());
+                    });
+
+                    if (enrichmentType.equals("SAMPLE")) {
+                        return structuralVariantService.getSampleCountInMultipleMolecularProfiles(molecularProfileIds, caseIds,
```

Comment:
How many case sets could you potentially be iterating over? Iteratively calling a service seems like it could be a good way to run into performance issues.

Reply:
It can be any, it is the number of groups in comparison. this would be the only way to do it as the response from `getSampleCountInMultipleMolecularProfiles` would be aggregated counts